packages/teleterm/src/ui/services/connectionTracker/connectionTrackerService.ts

```diff
@@ -64,10 +65,11 @@ export class ConnectionTrackerService extends ImmutableStore<ConnectionTrackerSt
     return this.state.connections.map(connection => {
       const { rootClusterUri, leafClusterUri } =
         this._trackedConnectionOperationsFactory.create(connection);
-      const cluster = this._clusterService.findCluster(
-        leafClusterUri || rootClusterUri
-      );
-      return { ...connection, clusterName: getClusterName(cluster) };
+      const clusterUri = leafClusterUri || rootClusterUri;
+      const clusterName =
+        getClusterName(this._clusterService.findCluster(clusterUri)) ||
+        routing.parseClusterName(clusterUri);
+      return { ...connection, clusterName };
```

Comment:
This whole thing can be removed once we implement gravitational/webapps.e#261 and the cluster name will be known upfront, right?

Then we won't need to do this whole map but instead set the cluster name once when the connection is created and then just return `this.state.connections`. The cluster name never changes.

Reply:
Generally yes, however I wanted to avoid saving that information if we can calculate it basing on `clusterUri`.

> The cluster name never changes.

It reminded mi about [that list](https://www.kalzumeus.com/2010/06/17/falsehoods-programmers-believe-about-names/).
> People’s names do not change. 

😄